src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -68,8 +85,9 @@ public ServiceBusJmsConnectionFactory(String connectionString, ServiceBusJmsConn
     public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBuilder, ServiceBusJmsConnectionFactorySettings settings) {
         this.builder = connectionStringBuilder;
         this.settings = settings;
-        this.initialize(connectionStringBuilder.getSasKeyName(),
-                connectionStringBuilder.getSasKey(),
+        this.setKeyValue(connectionStringBuilder.getSasKey());
```

Comment:
>setKeyValue [](http://example.com/codeflow?start=13&length=11)

nit: Any reason keyValue has a setter where as keyName is set here directly? Can we keep them consistent.

Reply:
not anymore it was to test other function I add but end up not working I will do another pass make sure I do another pass to make sure solution was clean